cpp/include/cudf_test/testing_main.hpp

```diff
@@ -191,22 +192,30 @@ inline auto make_stream_mode_adaptor(cxxopts::ParseResult const& cmd_opts)
   return adaptor;
 }
 
+/**
+ * @brief Should be called in every test program that uses rmm allocators since it maintains the
+ * lifespan of the rmm default memory resource. this function parses the command line to customize
+ * test behavior, like the allocation mode used for creating the default memory resource.
+ *
+ */
+inline void init_cudf_test(int argc, char** argv)
+{
+  // static lifetime to keep rmm resource alive till tests end
+  auto const cmd_opts                  = parse_cudf_test_opts(argc, argv);
+  [[maybe_unused]] static auto mr      = make_memory_resource_adaptor(cmd_opts);
+  [[maybe_unused]] static auto adaptor = make_stream_mode_adaptor(cmd_opts);
+}
+
 /**
  * @brief Macro that defines main function for gtest programs that use rmm
  *
- * Should be included in every test program that uses rmm allocators since
- * it maintains the lifespan of the rmm default memory resource.
  * This `main` function is a wrapper around the google test generated `main`,
- * maintaining the original functionality. In addition, this custom `main`
- * function parses the command line to customize test behavior, like the
- * allocation mode used for creating the default memory resource.
+ * maintaining the original functionality.
  */
-#define CUDF_TEST_PROGRAM_MAIN()                                            \
-  int main(int argc, char** argv)                                           \
-  {                                                                         \
-    ::testing::InitGoogleTest(&argc, argv);                                 \
-    auto const cmd_opts           = parse_cudf_test_opts(argc, argv);       \
-    [[maybe_unused]] auto mr      = make_memory_resource_adaptor(cmd_opts); \
-    [[maybe_unused]] auto adaptor = make_stream_mode_adaptor(cmd_opts);     \
-    return RUN_ALL_TESTS();                                                 \
+#define CUDF_TEST_PROGRAM_MAIN()            \
+  int main(int argc, char** argv)           \
+  {                                         \
+    ::testing::InitGoogleTest(&argc, argv); \
+    init_cudf_test(argc, argv);             \
+    return RUN_ALL_TESTS();                 \
```

Comment:
The large-strings tests provide there own fixture/main. 
Does the [large_strings_fixture.cpp](https://github.com/rapidsai/cudf/blob/1436cac9de8b450a32e71d5b779503e9a29edaa6/cpp/tests/large_strings/large_strings_fixture.cpp#L123) need additional changes as well?

Reply:
no it doesn't, although it would be ideal to use the init_cudf_test method instead of the cudf setup code already in large_string_fixture's main. it's just to allow cudftestutil consumers set up their tests differently as they won't necessarily know all the required setup. 
I'll make the change